test/test.js

```diff
@@ -162,4 +167,98 @@ describe('use mongodb 3 driver in a 2.x style', function() {
       assert(!trees.find(tree => (tree.ohmy !== true) || tree.leaves));
     });
   });
+  it('aggregation query works', function() {
+    return trees.aggregate([
+      {
+        $match: {
+          ohmy: true
+        }
+      }
+    ]).then(function(result) {
+      assert(result);
+      assert(Array.isArray(result));
+      assert(result.length);
```

Comment:
It seems like we should be testing for specific lengths where we do this assertion. Not a deal breaker, though.